src/back/hlsl/writer.rs

```diff
@@ -1441,6 +1566,180 @@ impl<'a, W: Write> Writer<'a, W> {
 
         Ok(())
     }
+
+    /// Helper function that write wrapped function for `Expression::ImageQuery`
+    /// https://docs.microsoft.com/en-us/windows/win32/direct3dhlsl/dx-graphics-hlsl-to-getdimensions
+    fn write_wrapped_image_query_functions(
```

Comment:
We should start thinking about how to split the HLSL backend into sub-modules. This isn't needed for this PR, but technically we could move all the image-related processing into a module in the future.